go/pull_request.go

```diff
@@ -627,22 +637,28 @@ func (h *PullRequestHandler) createCobraDocsPreviewPR(
 		return nil, errors.Wrapf(err, "Failed to run cobradocs sync script against %s:%s to %s for %s", remote, ref, op, pr.GetHTMLURL())
 	}
 
-	// Amend the commit to change the author to the bot.
-	if err := website.Commit(ctx, fmt.Sprintf("generate cobradocs against %s:%s", remote, ref), git.CommitOpts{
-		Author: botCommitAuthor,
-		Amend:  true,
-	}); err != nil {
-		return nil, errors.Wrapf(err, "Failed to amend commit author to %s for %s", op, pr.GetHTMLURL())
+	tree, commit, err := h.writeAndCommitTree(
+		ctx,
```

Comment:
not super important, but we should probably update the following comments `// 4.`, `// 5.`, ... since we added the write and commit tree step

Reply:
will do that as part of #77